smartsim/entity/model.py

```diff
@@ -291,7 +303,39 @@ def _set_colocated_db_settings(
         if hasattr(self.run_settings, "_prep_colocated_db"):
             self.run_settings._prep_colocated_db(common_options["cpus"])
 
+        if "limit_app_cpus" in common_options:
+            raise SSUnsupportedError(
+                "Pinning of app CPUs via limit_app_cpus is no supported. Modify RunSettings " +
+                "instead using the correct binding option for your launcher."
+            )
+
         # TODO list which db settings can be extras
+        cpus = common_options["cpus"]
+        # Deal with cases where the database should be pinned to cpus
+        # (1) if the user set a db_cpu_list, but not limit_db_cpus
+        if common_options["db_cpu_list"] and not common_options["limit_db_cpus"]:
+            logger.warning(
+                "limit_db_cpus is False, but db_cpu_list is not None. Setting limit_db_cpus=True"
+            )
+            common_options["limit_db_cpus"] = True
+        # (2) limit_db_cpus, but not db_cpu_list is specified automatically set
+        #     pin to the cpus 0:db_cpus-1
+        if common_options["limit_db_cpus"] and not common_options["db_cpu_list"]:
+            if cpus > 1:
+                cpu_list = f"0-{cpus-1}"
+            elif cpus == 1:
+                cpu_list = "0"
+            else:
```

Comment:
Should we add logic to handle the case were a user requests that the `db_cpu_list` contain more cpus than the number of requested cpus (e.g. `cpus == 1` and `cpu_list == "0,99"`)? Or should we chalk that up to "user error"?

Reply:
That's actually an allowed case because taskset just tells the application what CPUs it CAN be deployed in. So it's entirely possible for cpus=1 and then to have len(cpu_list)>1